src/test/java/com/google/cloud/resourcemanager/LienTest.java

```diff
@@ -0,0 +1,175 @@
+          .setRestrictions(LIEN_RESTRICTIONS)
+          .build();
+  private static final LienInfo LIEN_INFO1 = new LienInfo.BuilderImpl(LIEN_INFO).build();
+  private static final LienInfo LIEN_INFO2 = new LienInfo.BuilderImpl(LIEN_INFO).build();
+
+  private ResourceManager serviceMockReturnsOptions = createStrictMock(ResourceManager.class);
+  private ResourceManagerOptions mockOptions = createMock(ResourceManagerOptions.class);
+  private ResourceManager resourceManager;
+  private Lien expectedLien;
+  private Lien lien;
+
+  @Before
+  public void setUp() {
+    resourceManager = createStrictMock(ResourceManager.class);
+  }
+
+  @After
+  public void tearDown() {
+    verify(resourceManager);
+  }
+
+  private void initializeExpectedLien(int optionsCalls) {
+    expect(serviceMockReturnsOptions.getOptions()).andReturn(mockOptions).times(optionsCalls);
+    replay(serviceMockReturnsOptions);
+    expectedLien = new Lien(serviceMockReturnsOptions, new LienInfo.BuilderImpl(LIEN_INFO));
+  }
+
+  private void initializeLien() {
+    lien = new Lien(resourceManager, new LienInfo.BuilderImpl(LIEN_INFO));
+  }
+
+  @Test
+  public void testToBuilder() {
+    initializeExpectedLien(4);
+    replay(resourceManager);
+    compareLiens(expectedLien, expectedLien.toBuilder().build());
+  }
+
+  @Test
+  public void testBuilder() {
+    expect(resourceManager.getOptions()).andReturn(mockOptions).times(4);
+    replay(resourceManager);
+    Lien.Builder builder =
+        new Lien.Builder(new Lien(resourceManager, new LienInfo.BuilderImpl(LIEN_PARENT)));
+    Lien lien =
+        builder
+            .setName(LIEN_NAME)
+            .setCreateTime(LIEN_CREATE_TIME)
+            .setParent(LIEN_PARENT)
+            .setOrigin(LIEN_ORIGIN)
+            .setReason(LIEN_REASON)
+            .setRestrictions(LIEN_RESTRICTIONS)
+            .build();
+    assertEquals(LIEN_NAME, lien.getName());
+    assertEquals(LIEN_CREATE_TIME, lien.getCreateTime());
+    assertEquals(LIEN_ORIGIN, lien.getOrigin());
+    assertEquals(LIEN_REASON, lien.getReason());
+    assertEquals(LIEN_RESTRICTIONS, lien.getRestrictions());
+    assertEquals(LIEN_PARENT, lien.getParent());
+    assertEquals(resourceManager.getOptions(), lien.getResourceManager().getOptions());
+  }
+
+  @Test
+  public void testCreate() {
+    initializeExpectedLien(1);
+    expect(resourceManager.getOptions()).andReturn(mockOptions);
+    expect(resourceManager.createLien(LIEN_INFO)).andReturn(expectedLien);
+    replay(resourceManager);
+    initializeLien();
+    Lien actualLien = lien.create(LIEN_INFO);
+    compareLiens(LIEN_INFO, actualLien);
+    assertEquals(LIEN_NAME, actualLien.getName());
+    assertEquals(LIEN_CREATE_TIME, actualLien.getCreateTime());
+    assertEquals(LIEN_ORIGIN, actualLien.getOrigin());
+    assertEquals(LIEN_REASON, actualLien.getReason());
+    assertEquals(LIEN_RESTRICTIONS, actualLien.getRestrictions());
+    assertEquals(LIEN_PARENT, actualLien.getParent());
```

Comment:
You have already compared name, parent, restrictions, etc in `compareLiens()` method then why do you asserting them again here?